public/javascripts/controllers.js

```diff
@@ -1,692 +1,816 @@
+          size: 'lg',
+          scope: $scope
+        });
 
-    $http.post('/api/visualizations/save', {type: type, visualizations: $scope.visualizations})
-    .success(function(data){
-      if(!data.error){
-        $modalInstance.close('saved');
-      }
-    });
-  }
-})
+        $scope.modalInstance.opened.then(function(){
+
+          $scope.save = function(user){
+            $scope.cuErrorMessages = [];
+            $scope.cuError = false;
+
+            // Checks
+            if(typeof user.firstName === "undefined"){
+              $scope.cuErrorMessages.push('First name is required');
+              $scope.cuError = true;
+            }
 
-.controller('userOverviewCtrl', function($scope){
+            if(typeof user.lastName === "undefined"){
+              $scope.cuErrorMessages.push('Last name is required');
+              $scope.cuError = true;
+            }
+
+            if(typeof user.email === "undefined"){
+              $scope.cuErrorMessages.push('Email is required');
+              $scope.cuError = true;
+            }
+
+            if(typeof user.password === "undefined"){
+              $scope.cuErrorMessages.push('Password is invalid. Make sure the password is at least 4 characters long.');
+              $scope.cuError = true;
+            }
+
+            if(typeof user.password2 === "undefined"){
+              $scope.cuErrorMessages.push('Make sure the passwords match.');
+              $scope.cuError = true;
+            }
 
-  $scope.rowCollection = [
-    {id: 1, firstName: 'Laurent', lastName: 'Renard', email: 'whatever@gmail.com', role: 'Coffee Boy'},
-    {id: 2, firstName: 'Blandine', lastName: 'Faivre', email: 'oufblandou@gmail.com', role: 'Evidence Searcher'},
-    {id: 3, firstName: 'Francoise', lastName: 'Frere', email: 'raymondef@gmail.com', role: 'Evidence Officer'}
-  ];
+            if($scope.cuError === false){
+
+              if(user.password !== user.password2){
+                $scope.cuErrorMessages.push('Password doesn\'t match.');
+                $scope.cuError = true;
+
+              }else if(user.password === user.password2){
+                delete user.password2;
+                var addUser = true;
+
+                // Check if user exists
+                angular.forEach(users, function(value, key){
+                  var u = value._source;
+                  if(user.email == u.email){
+                    $scope.cuErrorMessages.push('An user with this email address already exists.');
+                    $scope.cuError = true;
+                    addUser = false;
+                  }
+                });
+
+                // Save user
+                if(addUser) {
+                  $http.post('/api/save_user', user)
+                      .success(function (data) {
+                            if (typeof data.error !== 'undefined') {
+                              $scope.error.push(data.error.message);
+                              $scope.isError = true;
```

Comment:
Please see my comment on line 686 for a proper solution for using this extra `bool`. This counts for all checks.